lib/active_model/serializer/adapter.rb

```diff
@@ -86,7 +87,11 @@ def meta_key
       end
 
       def root
-        serializer.json_key.to_sym if serializer.json_key
+        if serializer.json_key
+          serializer.json_key.to_sym
+        else
+          serializer.root
```

Comment:
Sorry, `serializer.root` is what I meant.. (`serializer.json_key` can't be nil.to_sym because it's the consequence to the truthy json_key